streamDevice/src/StreamEpics.cc

```diff
@@ -944,12 +944,13 @@ getFieldAddress(const char* fieldname, StreamBuffer& address)
     else
     {
         // FIELD in this record or VAL in other record
-        char fullname[PVNAME_SZ + 1];
-        sprintf(fullname, "%s.%s", name(), fieldname);
+        size_t fullnameLen = PVNAME_SZ + 1;
```

Comment:
You may be using a gcc specific feature here.  https://gcc.gnu.org/onlinedocs/gcc/Variable-Length.html#Variable-Length

Better to do

```c
char fullname[PVNAME_SZ+1];
const size_t fullnameLen = sizeof(fullname);
```

In addition, since field names are rarely more than 4 charactors, a buffer size of PVNAME_SZ+6 is a better fit.